tests/integration/pilot/nodelocal/node_local_test.go

```diff
@@ -0,0 +1,174 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodelocal
+
+import (
+	"context"
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	pilotcommont "istio.io/istio/pkg/test/framework/components/echo/common"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/tests/integration/pilot/common"
+)
+
+const (
+	nodeLocalSvcA = "node-local-svc-a"
+	nodeLocalSvcB = "node-local-svc-b"
+	nodeLocalSvcC = "node-local-svc-c"
+)
+
+var (
+	i    istio.Instance
+	apps = &common.EchoDeployments{}
+)
+
+// TestMain defines the entrypoint for pilot tests using a standard Istio installation.
+// just use for this test.
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		Label(label.CustomSetup).
+		RequireMinClusters(2).
+		RequireMinVersion(21).
+		Setup(istio.Setup(&i, enableK8sInternalTrafficPolicy)).
+		Setup(deployEchos).
+		Run()
+}
+
+func enableK8sInternalTrafficPolicy(_ resource.Context, cfg *istio.Config) {
+	cfg.ControlPlaneValues = `
+values:
+  pilot:
+    env:
+      PILOT_ENABLE_KUBERNETES_INTERNAL_TRAFFIC_POLICY: true
+`
+}
+
+func deployEchos(t resource.Context) error {
+	var err error
+	var echos echo.Instances
+	// Create a new namespace in each cluster.
+	apps.Namespace, err = namespace.New(t, namespace.Config{
+		Prefix: "node-local",
+		Inject: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	primary := t.Clusters().Primaries()[0]
```

Comment:
I wonder if we should add a node field to our echo responses for validation, and store the node for the caller. I have a feeling equating cluster with node will break in other envs besides our kind env in CI.

Reply:
Each cluster has only one node. I wonder if it can be distinguished according to the cluster. The same cluster can be accessed, but other clusters fail to access. If we change the number of test cluster nodes in the future, this method will fail.
Next，i will try to add node name in the response and instance